network/p2p/libp2pUtils.go

```diff
@@ -98,37 +101,99 @@ func filterStream(host host.Host, targetID peer.ID, protocol core.ProtocolID, di
 	}
 
 	return addr, nil
 }
 
-// NodeAddressFromIdentity returns the libp2p.NodeAddress for the given flow.identity
-func NodeAddressFromIdentity(flowIdentity flow.Identity) (NodeAddress, error) {
-
+// networkingInfo returns ip, port, libp2p public key of the identity.
+func networkingInfo(identity flow.Identity) (string, string, crypto.PubKey, error) {
 	// split the node address into ip and port
-	ip, port, err := net.SplitHostPort(flowIdentity.Address)
+	ip, port, err := net.SplitHostPort(identity.Address)
 	if err != nil {
-		return NodeAddress{}, fmt.Errorf("could not parse address %s: %w", flowIdentity.Address, err)
+		return "", "", nil, fmt.Errorf("could not parse address %s: %w", identity.Address, err)
 	}
 
 	// convert the Flow key to a LibP2P key
-	lkey, err := publicKey(flowIdentity.NetworkPubKey)
+	lkey, err := publicKey(identity.NetworkPubKey)
+	if err != nil {
+		return "", "", nil, fmt.Errorf("could not convert flow key to libp2p key: %w", err)
+	}
+
+	return ip, port, lkey, nil
+}
+
+// MultiAddressStr receives a node ip and port and returns
+// its corresponding Libp2p MultiAddressStr in string format
+// in current implementation IP part of the node address is
+// either an IP or a dns4.
+// https://docs.libp2p.io/concepts/addressing/
+func MultiAddressStr(ip, port string) string {
+	parsedIP := net.ParseIP(ip)
+	if parsedIP != nil {
+		// returns parsed ip version of the multi-address
+		return fmt.Sprintf("/ip4/%s/tcp/%s", ip, port)
+	}
+	// could not parse it as an IP address and returns the dns version of the
+	// multi-address
+	return fmt.Sprintf("/dns4/%s/tcp/%s", ip, port)
+}
+
+// IPPortFromMultiAddress returns the IP/hostname and the port for the given multi-addresses
+// associated with a libp2p host
+func IPPortFromMultiAddress(addrs ...multiaddr.Multiaddr) (string, string, error) {
+
+	var ipOrHostname, port string
+	var err error
+
+	for _, a := range addrs {
+		// try and get the dns4 hostname
+		ipOrHostname, err = a.ValueForProtocol(multiaddr.P_DNS4)
+		if err != nil {
+			// if dns4 hostname is not found, try and get the IP address
+			ipOrHostname, err = a.ValueForProtocol(multiaddr.P_IP4)
+			if err != nil {
+				continue // this may not be a TCP IP multiaddress
+			}
+		}
+
+		// if either IP address or hostname is found, look for the port number
+		port, err = a.ValueForProtocol(multiaddr.P_TCP)
+		if err != nil {
+			// an IPv4 or DNS4 based multiaddress should have a port number
+			return "", "", err
+		}
+
+		//there should only be one valid IPv4 address
+		return ipOrHostname, port, nil
+	}
+	return "", "", fmt.Errorf("ip address or hostname not found")
+}
+
+func generateProtocolID(rootBlockID string) protocol.ID {
+	return protocol.ID(FlowLibP2PProtocolIDPrefix + rootBlockID)
+}
+
+// PeerAddressInfo generates the libp2p peer.AddrInfo for an identity given its node address.
+func PeerAddressInfo(identity flow.Identity) (peer.AddrInfo, error) {
```

Comment:
redundant with `PeerInfoFromID`

Reply:
Fixed in https://github.com/onflow/flow-go/pull/211/commits/ea1598a5c3cffa8d521f467aab03562db2da8c5e